src/idiomatic/leveraging-the-type-system/borrow-checker-invariants/phantomdata-01-types.md

```diff
@@ -0,0 +1,117 @@
+---
+minutes: 15
+---
+
+# PhantomData 1/2: Type-level tagging
+
+<!-- dprint-ignore-start -->
+```rust,editable
+// use std::marker::PhantomData;
+
+pub trait ChatUser {}
+pub trait ChatModerator {}
+pub trait ChatAdmin {}
+
+pub struct UserId(u64);
+impl ChatUser for UserId { /* ... */ }
+
+pub struct PatronId(u64);
+impl ChatUser for PatronId { /* ... */ }
+
+pub struct ModeratorId(u64);
+impl ChatUser for ModeratorId { /* ... */ }
+impl ChatModerator for ModeratorId { /* ... */ }
+
+pub struct AdminId(u64);
+impl ChatUser for AdminId { /* ... */ }
+impl ChatModerator for AdminId { /* ... */ }
+impl ChatAdmin for AdminId { /* ... */ }
+
+// And so on ...
+fn main() {}
+```
+<!-- dprint-ignore-end -->
+
+<details>
+
+- Problem: We want to use the newtype pattern to differentiate permissions, but
+  we're duplicating identical implementations for identical data.
+
+- Motivation: We want to be able to "tag" structures with different type
+  parameters as a way to tell them apart or pass on lifetime information to
+  them.
+
+  See: [Typestate Generics](../typestate-pattern/typestate-generics.md) for
+  instances of telling apart different data relevant to stages of an algorithm
+  with type parameter differences.
+
+  In practice, these "tags" tend to be zero-sized types. What they mean will
```

Comment:
This sentence would land better once we are already showing the new implementation. (move it down)